packages/spark-core/src/lib/storage/decorators.js

```diff
@@ -98,14 +97,39 @@ export function waitForValue(key) {
   };
 }
 
+
+const inited = new Set();
+
+/**
+ * finds a means of identitying the `target` param passed to
+ * `prepareInitialize()`. When possible, avoids duplicate `init()` calls if
+ * namespaces collide
+ *
+ * @param {Object|Constructor} target
+ * @private
+ * @returns {String|Constructor}
+ */
+function identifyTarget(target) {
+  if (target.getNamespace) {
+    return target.getNamespace();
+  }
+
+  if (target.namespace) {
+    return target.namespace;
+  }
```

Comment:
safety. `getNamespace` will probably never get called because `identifyTarget` gets the object that's passed to `SparkPlugin.extend()` rather than the result. That said, `getNamespace()` is the right way to get a namespace from a plugin, so I'm just planning for a potential future.